lib/src/clojure_lsp/dep_graph.clj

```diff
@@ -0,0 +1,347 @@
+;;  (':from-internal?' boolean)?
+;;  (':from-langs' #<lang*>)?}
+
+;; :dependencies is a multiset of the namespaces that this namespace depends on.
+;; It's a multiset because a dependency may be required several times, either
+;; with different aliases, or because the dependent namespace is defined across
+;; several files. We keep a multiset because removing one of these duplicates
+;; doesn't erase the dependency/dependent relationship. If a namespace has no
+;; dependencies, the :dependencies will either be absent from the
+;; dep-graph-item, or be an empty multiset. The system doesn't make guarantees
+;; about which it will be.
+
+;; :dependents is a multiset of the namespaces that depend on this namespace.
+;; May be either empty or absent if namespace isn't depended on by other
+;; namespaces.
+
+;; If the code had `(ns aaa (require [bbb :as b]))`, then `'bbb` would be in
+;; `'aaa`'s :dependencies, and `'aaa` would be in `'bbb`'s :dependents.
+
+;; :aliases is a multiset of the symbols that are used to refer to this
+;; namespace. If the code had `(:require [aaa :as a])`, then `'aaa` would
+;; include `'a` in its aliases. Unaliased requires like `(:require [aaa])` will
+;; include `nil` in their aliases. May be either empty or absent if a namespace
+;; isn't required by other namespaces.
+
+;; :files is a set of the files in which the namespace is defined. This can be
+;; either absent or empty if the namespace is required but never defined, either
+;; as a syntax error or via :as-alias.
+
+;; :internal? is true when the namespace is defined in any internal file, falsy
+;; otherwise. May be absent if the namespace is required but never defined, as
+;; for :files.
+
+;; :from-internal? is true when the namespace is required by any internal
+;; namespace, falsy otherwise. May be absent if the namespace isn't required by
+;; other namespaces.
+
+;; :from-langs is a multiset of the langs of the :namespace-usages or files (if the
+;; :namespace-usages don't have langs) that use this namespace. May be either
+;; empty or absent if the namespace isn't required by other namespaces.
+
+;; file-meta-item =
+;; {(:namespaces #{ns*})?
+;;  :internal? boolean
+;;  :langs #{lang+}}
+
+;; :namespaces is a set of the namespaces defined by this file. May be either
+;; empty or absent if the file doesn't define any namespaces.
+
+;; :internal? is whether this file is internal to the project.
+
+;; :langs is a set of the langs used by this file.
+
+;; file-meta = {(filename file-meta-item)*}
+;; dep-graph = {(ns dep-graph-item)*}
+;; db = {':dep-graph' dep-graph
+;;       ':file-meta' file-meta}
+
+(comment
+  ;; Example
+  (require '[clojure-lsp.db :as db])
+  (-> @db/db*
+      :dep-graph
+      (select-keys '[clojure-lsp.main
+                     clojure.tools.cli]))
+  '{clojure-lsp.main  {:dependents     {clojure-lsp.main-test 1}
+                       :aliases        {main 1}
+                       :from-internal? true
```

Comment:
Not sure it's clear that from-internal means that this ns used by a ns internally, maybe `used-internally?`

Reply:
`internal-dependency`... naming is hard

Reply:
I want to keep `:from-internal?` consistent with `:from-langs`. I'm going to go with `:dependents-internal?` and `:dependents-langs`. It's still not perfectly clear, but it emphasizes that it's additional data about the dependents. And as you said naming is hard.